server/src/main/java/org/opensearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -124,72 +124,45 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
         Weight subQueryWeight = subQuery.createWeight(searcher, subQueryScoreMode, 1.0f);
 
         return new Weight(this) {
+
             @Override
             public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOException {
-                final Weight weight = this;
+                ScorerSupplier subQueryScorerSupplier = subQueryWeight.scorerSupplier(context);
+                if (subQueryScorerSupplier == null) {
+                    return null;
+                }
```

Comment:
Ensuring the `subQueryScorerSupplier` is non-null here helps make the `ScorerSupplier` implementation cleaner and easier.